src/main/java/edu/harvard/iq/dataverse/api/EditDDI.java

```diff
@@ -110,12 +110,16 @@ public Response edit(@Context ContainerRequestContext crc, InputStream body, @Pa
         Map<Long, VariableMetadata> mapVarToVarMet = new HashMap<Long, VariableMetadata>();
         Map<Long,VarGroup> varGroupMap = new HashMap<Long, VarGroup>();
         try {
-            readXML(body, mapVarToVarMet,varGroupMap);
+            if (!readXML(body, mapVarToVarMet, varGroupMap)) {
+                return error(Response.Status.BAD_REQUEST, "bad xml file");
```

Comment:
to help with debugging, could the error text for this be different from when there's an XML exception? Is this error "xml missing fields" or "xml isn't valid DDI" or ?

Reply:
@qqmyers I updated as you suggested. Now it catches the original exception and returns 400 error with a message  frequency or weighted frequency being declared but missing.